packages/expo-keep-awake/src/index.ts

```diff
@@ -1,20 +1,41 @@
 import { useEffect } from 'react';
 
-import ExpoKeepAwake from './ExpoKeepAwake';
+import ExpoKeepAwake, { isAvailable } from './ExpoKeepAwake';
+import { KeepAwakeEventType } from './ExpoKeepAwake.types';
 
 const ExpoKeepAwakeTag = 'ExpoKeepAwakeDefaultTag';
 
-export function useKeepAwake(tag: string = ExpoKeepAwakeTag): void {
+function useKeepAwake(tag: string = ExpoKeepAwakeTag): void {
   useEffect(() => {
     activateKeepAwake(tag);
-    return () => deactivateKeepAwake(tag);
+    return () => {
+      deactivateKeepAwake(tag);
+    };
   }, [tag]);
 }
 
-export function activateKeepAwake(tag: string = ExpoKeepAwakeTag): void {
-  if (ExpoKeepAwake.activate) ExpoKeepAwake.activate(tag);
+async function activateKeepAwake(tag: string = ExpoKeepAwakeTag): Promise<void> {
+  if (isAvailable() && ExpoKeepAwake.activate) {
+    await ExpoKeepAwake.activate(tag);
+  }
```

Comment:
```suggestion
  if (!(await isAvailableAsync())) {
    if (__DEV__) {
      console.warn('KeepAwake.activateKeepAwake() was invoked on a device that doesn't support the KeepAwake API.');
    }
    return;
  }
  if (ExpoKeepAwake.activate) {
    await ExpoKeepAwake.activate(tag);
  }
```

Let's warn in dev mode that the API has been misused (we do this in AuthSession API)

Reply:
Apply the same to all methods